src/auth/auth.service.ts

```diff
@@ -30,7 +30,13 @@ export class AuthService {
     };
   }
 
-  signUp(signUpDto: SignUpDto) {
-    return this.userService.create(signUpDto);
+  signIn(user: User) {
+    return this.generateUserWithJwt(user);
+  }
+
+  async signUp(signUpDto: SignUpDto) {
+    const user = await this.userService.create(signUpDto);
+
+    return this.generateUserWithJwt(user);
```

Comment:
in fact for singleton pattern, we import user module into here, and use function userService.create